.github/workflows/codeql.yml

```diff
@@ -7,29 +7,39 @@ on:
       - development
       - "!dependabot/**"
   pull_request:
-    # The branches below must be a subset of the branches above
     branches:
       - master
       - development
+      - "!dependabot/**"
   schedule:
     - cron: "0 0 * * 0"
+  workflow_dispatch:
 
 jobs:
   analyze:
     name: Analyze
     runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
 
     steps:
-      - name: Checkout repository
+      - name: Clone repository
         uses: actions/checkout@v4.2.2
-      # Initializes the CodeQL tools for scanning.
+        with:
+          persist-credentials: false
+
       - name: Initialize CodeQL
         uses: github/codeql-action/init@v3
```

Comment:
Should we use the tag here too? I notice you guys have switched to the specific tags for other actions. I personally don't like this approach because it's unnecessary noise for GitHub actions, but asking for consistency.

Reply:
Yes please.